src/interpreter_funcs/voxel_boolean_union.rs

```diff
@@ -0,0 +1,261 @@
+use std::error;
+use std::f32;
+use std::fmt;
+use std::ops::Bound;
+use std::sync::Arc;
+
+use nalgebra::Vector3;
+
+use crate::analytics;
+use crate::bounding_box::BoundingBox;
+use crate::interpreter::{
+    BooleanParamRefinement, Float3ParamRefinement, Func, FuncError, FuncFlags, FuncInfo,
+    LogMessage, ParamInfo, ParamRefinement, Ty, UintParamRefinement, Value,
+};
+use crate::mesh::voxel_cloud::{self, ScalarField};
+
+const VOXEL_COUNT_THRESHOLD: u32 = 50000;
+
+#[derive(Debug, PartialEq)]
+pub enum FuncBooleanUnionError {
+    WeldFailed,
+    EmptyScalarField,
+    VoxelDimensionsZero,
+    TooManyVoxels(u32, f32, f32, f32),
+}
+
+impl fmt::Display for FuncBooleanUnionError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            FuncBooleanUnionError::WeldFailed => write!(
+                f,
+                "Welding of separate voxels failed due to high welding proximity tolerance"
+            ),
+            FuncBooleanUnionError::EmptyScalarField => write!(
+                f,
+                "Scalar field from input meshes or the resulting mesh is empty"
+            ),
+            FuncBooleanUnionError::VoxelDimensionsZero => write!(f, "One or more voxel dimensions are zero"),
+            FuncBooleanUnionError::TooManyVoxels(max_count, x, y, z) => write!(
+                f,
+                "Too many voxels. Limit set to {}. Try setting voxel size to [{:.3}, {:.3}, {:.3}] or more.",
+                max_count, x, y, z
+            ),
+        }
+    }
+}
+
+impl error::Error for FuncBooleanUnionError {}
+
+pub struct FuncBooleanUnion;
+
+impl Func for FuncBooleanUnion {
+    fn info(&self) -> &FuncInfo {
+        &FuncInfo {
+            name: "Union",
+            description: "BOOLEAN UNION OF VOXEL CLOUDS FROM TWO MESH GEOMETRIES\n\
+            \n\
+            Converts the input mesh geometries into voxel clouds, then performs \
+            boolean union of the first to second voxel clouds and eventually \
```

Comment:
Likewise: maybe we should be saying "first and second" here, because union commutes.